arch/armv7a/jmp.S

```diff
@@ -6,90 +6,107 @@
  *
  * _setjmp, _longjmp, setjmp, sigsetjmp
  *
- * Copyright 2018 Phoenix Systems
- * Author: Kamil Amanowicz
+ * Copyright 2018, 2019 Phoenix Systems
+ * Author: Kamil Amanowicz, Aleksander Kaminski, Jakub Sarzynski
  *
  * This file is part of Phoenix-RTOS.
  *
  * %LICENSE%
  */
 
+ /* setjmp env buffer description
+  * Offset	Description
+  * 0		Signal mask flag - if != 0 then next value stores a signal mask
+  * 4		Signal mask
+  * 8		SP register
+  * 12-44	r4-r11 registers
+  * 44		LR register
+  * 48-80	d8-d15 registers
+  * 80		fpscr register
+  */
+
 #define __ASSEMBLY__
 
 .thumb
 .syntax unified
 
-.globl _setjmp
-.type _setjmp, %function
-_setjmp:
-	/* set jump address and stack pointer */
-	ldr r1, =(1f + 1)
-	mov r2, sp
-	/* store everythig in jmpbuf */
-	stm r0, {r1, r2, r4, r5, r6, r7, r8, r9, r10, r11, lr}
+
+.globl sigsetjmp
+.type sigsetjmp, %function
+sigsetjmp:
+	/* Store signal mask if savesigs != 0 */
+	mov r2, #0
+	cbz r1, 1f
+
+	/* Get signal mask */
+	/* Stack must be aligned to 8 bytes */
+	push {r0, r1, r3, lr}
 	mov r0, #0
```

Comment:
```
movs r0, #0
or 
mov r0, r2 /* r2 contains 0 */
```

Reply:
Again, I see no reason why should s produce different instruction encoding

Reply:
You can use the second suggestion:
```
mov r0, r2 /* r2 contains 0 */
```